dynoid/dynoid.go

```diff
@@ -5,43 +5,228 @@ import (
 	"encoding/base64"
 	"encoding/json"
 	"fmt"
+	"log/slog"
+	"os"
+	"path"
 	"strings"
 	"sync"
 
 	"github.com/coreos/go-oidc/v3/oidc"
 )
 
-type IssuerCallback func(string) error
+const (
+	defaultAudience = "heroku"
 
-type Verifier struct {
-	IssuerCallback IssuerCallback
-	config         *oidc.Config
-	providers      sync.Map
+	//nolint: gosec
+	defaultTokenPath = "/etc/heroku/dyno_id_token"
+)
+
+// Returned by an IssuerCallback getting an issuer it doesn't trust
+type UntrustedIssuerError struct {
+	Issuer string
 }
 
-func New(clientID string) *Verifier {
-	return &Verifier{config: &oidc.Config{ClientID: clientID}}
+func (e *UntrustedIssuerError) Error() string {
+	return fmt.Sprintf("untrusted issuer: %v", e.Issuer)
+}
+
+// Returned when the token doesn't match the expected format
+type MalformedTokenError struct {
+	Err error
```

Comment:
We already define `Unwrap()` so this doesn't need to be exported:

```suggestion
	err error
```

Reply:
Updated in 1910815